src/init/mod.rs

```diff
@@ -61,25 +62,28 @@ pub fn print_logo() {
 #[cfg(not(target_os = "macos"))]
 fn mount_vfs(source_name: &str, target_name: &str, fstype: &str) {
     info!("Mounting {} as type {}", target_name, fstype);
-    unsafe {
-        // CString constructor ensures the trailing 0byte, which is required by libc::mount
-        let src_c_ctr = CString::new(source_name).unwrap();
-        let target_name_c_ctr = CString::new(target_name).unwrap();
-        let fstype_c_ctr = CString::new(fstype).unwrap();
-        let options_c_ctr = CString::new("").unwrap();
-
-        let ret = libc::mount(
-            src_c_ctr.as_ptr() as *const i8,
-            target_name_c_ctr.as_ptr() as *const i8,
-            fstype_c_ctr.as_ptr() as *const i8,
+
+    // CString constructor ensures the trailing 0byte, which is required by libc::mount
+    let src_c_ctr = CString::new(source_name).unwrap();
```

Comment:
If I got it correctly, we are thinking about throwing warnings if an unwrap is used. Maybe we could use this PR to also iterate on that? 

What is your recommendation on how to handle Result return types?

Reply:
To give you a bit more context: 

A naive example: 
https://github.com/aurae-runtime/auraed/blob/2b274806673ed958fbd5ac799cc7710cf463b758/src/init/network.rs#L82

Reply:
Regarding the unwraps, in this case, I would change them to `expect` calls since the fn is private and we are controlling the inputs, ensuring the inputs do not contain an internal 0 bytes (what the [docs](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.new) indicate would cause and error). If they are intended to be made pub or maybe even pub(crate) at some point, I would take a different approach (have the function return a `Result`).

For handling `Result` return types in general, I try to always choose handle what you can, and bubble up errors that you can't. I like the let it crash mentality (hopefully in fewer and fewer cases as code is iterated) so you don't have silent errors, but I don't know the appropriateness of that approach for this project. For context on me, all of the actually system stuff is new to me (I installed a linux vm for the first time to get the project to build), so I can't make an informed decision.

So, keeping in mind that I don't know what links or sriov are, looking at the two examples...

On the first, I'm not super comfortable with the Ok case having a warn (seems like that hides potential errors in the logs), but I like that the errors are bubbled up (we can replace anyhow with custom errors later).

On the second, it seems to fail silently (almost, there is the logs), and allow the program to continue, with the exception for failure to write to the file system, where it panics. Mixing the two approaches is confusing to me. If fail and continue is ok, I would still want a way to communicate that to the caller. For example, if the ultimate value of limit is meaningful to the caller, return `Option<u16>` where success is `Some(limit)` and failure is `None`. If failure is an error, then the fn should return a `Result`.

Reply:
Oh, and regarding changing `unwrap` to `expect`, can we assume the inputs will remain controlled?

Reply:
Thank you, this is very helpful! I will keep your recommendation in mind for my next PRs.

Yes, we can safely assume that the input is controlled. There might be a use case for exposing a gRPC api for mounting - @MalteJ @kris-nova? 
However, I would see it as the responsibility of a higher level function to safely call mount_vfs

Reply:
Great, I will update the pr...but first, I need breakfast, lol